src/storage/src/hummock/store/version.rs

```diff
@@ -353,30 +552,84 @@ impl HummockReadVersion {
             }
 
             VersionUpdate::CommittedSnapshot(committed_version) => {
-                let max_committed_epoch = committed_version.max_committed_epoch();
-                self.committed = committed_version;
+                let cleaned_ssts = self.clean_staging_ssts(committed_version);
 
-                {
-                    // TODO: remove it when support update staging local_sst
-                    self.staging
-                        .imm
-                        .retain(|imm| imm.min_epoch() > max_committed_epoch);
+                if let Some(committed_index) = self.committed_index.as_deref() {
+                    self.committed_index = Some(Arc::new(
+                        committed_index.apply(self.committed(), cleaned_ssts),
+                    ));
+                }
+            }
+        }
+    }
 
-                    self.staging
-                        .merged_imm
-                        .retain(|merged_imm| merged_imm.min_epoch() > max_committed_epoch);
+    fn clean_staging_ssts(
+        &mut self,
+        committed_version: PinnedVersion,
+    ) -> Vec<(HummockEpoch, Vec<HummockSstableObjectId>)> {
+        let old_max_committed_epoch = self.committed().max_committed_epoch();
+        self.committed = committed_version;
+        let new_max_committed_epoch = self.committed().max_committed_epoch();
+
+        let mut min_acceptable_staging_epoch = old_max_committed_epoch + 1;
+        if let Some(pruned_version) = self.committed_index.as_deref() && pruned_version.scope() > min_acceptable_staging_epoch {
+            min_acceptable_staging_epoch = pruned_version.scope();
```

Comment:
when pruned_version.scope() would be greater than the minimum acceptable staging epoch?

Reply:
> when pruned_version.scope() would be greater than the minimum acceptable staging epoch?

IMO, this won't happen.